File: ONLINE_CODING_TEST/school.programmers/lvl_1/86491_ 최소직사각형.java
class Solution {
  public int solution(int[][] sizes) {
    int maxW = Integer.MIN_VALUE;
    int maxH = Integer.MIN_VALUE;
    // {w, h}
    for (int i = 0; i < sizes.length; i++) {
      if (sizes[i][0] < sizes[i][1]) {
        sizes[i][0] ^= sizes[i][1];
        sizes[i][1] ^= sizes[i][0];
        sizes[i][0] ^= sizes[i][1];
      }
      maxW = Math.max(maxW, sizes[i][0]);
      maxH = Math.max(maxH, sizes[i][1]);
    } 
    return maxW * maxH;
  }
}

File: ONLINE_CODING_TEST/school.programmers/lvl_1/86491_ 최소직사각형.js
function solution(sizes) {
  let maxW = Number.MIN_SAFE_INTEGER;
  let maxH = Number.MIN_SAFE_INTEGER;
  // [w, h]
  for (let size of sizes) {
    if (size[0] < size[1]) {
      size[0] ^= size[1];
      size[1] ^= size[0];
      size[0] ^= size[1];
    }
    maxW = Math.max(maxW, size[0]);
    maxH = Math.max(maxH, size[1]);
  }
  return maxW * maxH;
}